### PR TITLE
Base version - CRM migrations-1.0.0

### PR DESCRIPTION
**Pull request by Prateek13727**

### CODE REVIEW

File: src/components/styles/abstract/_colorPalette.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+//border-colors
+$border-light: #f2f2f2;
+$border-dark: #828282;
+
+//background
+$bg-blue: #eef2ff;
+$bg-grey: #f3f3f3;
+$bg-transparent: rgba(255, 255, 255, 0.75);
+
```
Comment (ashutheone) [inline src/components/styles/abstract/_colorPalette.scss:18]: Top section of color palette is not generic. It is specific to a use-case.
Either these should be renamed or moved to repo where they are used.

'status colors' section has names which are suitable for a top level palette.

File: src/components/styles/abstract/_mixins.scss [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  line-height: 1;
+  letter-spacing: normal;
+}
+@mixin porter-header-background {
+  background-image: linear-gradient(
+    to right,
+    $color-blue-dark,
+    $color-blue-light
+  );
+}
```
Comment (ashutheone) [inline src/components/styles/abstract/_mixins.scss:15]: These don't belong here.

Comment (ashutheone) [reply src/components/styles/abstract/_mixins.scss:15]: Or should be renamed

File: src/components/styles/base/_base.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+iframe {
+  border-top: none;
+}
+
+a {
+  cursor: pointer;
+}
+
```
Comment (ashutheone) [inline src/components/styles/base/_base.scss:20]: Not necessarily

File: src/components/styles/base/_utilities.scss [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+  border-right: 1px solid $color-off-white-2;
+}
+
+//position
+.u-position-sticky {
+  position: sticky;
+  top: 0px;
+  z-index: 3;
+}
+
```
Comment (ashutheone) [inline src/components/styles/base/_utilities.scss:270]: Use case specific

File: src/components/styles/components/_header.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+.header__h3 {
+  text-align: center;
+  font-size: $font-size-base;
+}
+
+.porter-logo {
+  height: 1.6rem;
+  margin-bottom: 1.2rem;
+}
```
Comment (ashutheone) [inline src/components/styles/components/_header.scss:22]: Isn't this for workspace?

File: src/components/styles/main.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
+@import "abstract/variables";
```
Comment (ashutheone) [inline src/components/styles/main.scss:1]: Where is main.scss used?

Comment (ashutheone) [reply src/components/styles/main.scss:1]: This way it will be a bulky import always. We need to split it up.

File: src/entities/OptionList.re [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module type SelectListConfig = {type uniqueIdType;};
```
Comment (ashutheone) [inline src/entities/OptionList.re:1]: What is this for? This is probably not similar to other entities.

File: src/helper-modules/csv-parser/CsvRow.re [hunk 8]
```diff
@@ -0,0 +1 @@
+type t = ReMapUtils.StringMap.t(string);
```
Comment (ashutheone) [inline src/helper-modules/csv-parser/CsvRow.re:1]: Can we make a types folder for this and others below?

File: src/helper-modules/reasonML/FlatResult.re [hunk 9]
```diff
@@ -0,0 +1 @@
+let let_ = Belt.Result.flatMap;
```
Comment (ashutheone) [inline src/helper-modules/reasonML/FlatResult.re:1]: Remove duplicate

File: src/helper-modules/sentry/Sentry.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+import * as Sentry from '@sentry/browser';
+var SENTRY_FRONTEND_DSN = 'https://4ce8c888a1d349428e0dfeb0cb325a71@sentry.io/1408218';
+let ignoreErrors = [
+  /Cannot read property 'getReadModeExtract' of undefined/i,
+  /Cannot read property 'getReadModeRender' of undefined/i,
+  /Cannot read property 'getReadModeConfig' of undefined/i,
+  /ResizeObserver loop limit exceeded/i,
+  /Unexpected end of input/i,
+  /Cannot read property \'\d+\' of null/i
```
Comment (ashutheone) [inline src/helper-modules/sentry/Sentry.js:9]: Remove these

File: src/utils/ArrayUtils.re [hunk 11]
```diff
@@ -0,0 +1,4 @@
+let firstN = (range: int, inputArray: array('a)): array('a) =>
```
Comment (ashutheone) [inline src/utils/ArrayUtils.re:1]: is this needed?

File: src/utils/FormValidationUtils.re [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+     );
+  result^;
+};
+
+let validateMsisdn = (msisdn: string): Belt.Result.t(string, string) => {
+  Js_re.test_([%re "/^(([6-9](\d{12}))|([0]))$/i"], msisdn)
+    ? Belt.Result.Ok(msisdn) : Belt.Result.Error("Enter valid msisdn");
+};
+
+let validateCallerNumber = (caller_number: string): Belt.Result.t(string, string) => {
```
Comment (ashutheone) [inline src/utils/FormValidationUtils.re:28]: rename to validateMobileNUmber

File: src/utils/ValidationHelper.re [hunk 13]
```diff
@@ -0,0 +1,29 @@
+module MobileNumber = {
```
Comment (ashutheone) [inline src/utils/ValidationHelper.re:1]: This is a repeat

**Review by ashutheone — CHANGES_REQUESTED** [review]

File: src/components/custom/cards/card_deprecated/CardComponentV2.re [hunk 14]
```diff
@@ -0,0 +1,10 @@
+[@react.component]
+let make = (~cardData: TableDataType.t) =>
+  <div className="card-component__main-container">
+    <CardHeaderComponent
+      title={cardData.title}
+      actionBtn={cardData.actionBtnTop}
+    />
+    <CardTableComponent tableData={cardData.tableData} />
+    <CardFooterComponent actionBtn={cardData.actionBtnBot} />
+  </div>;
```
[the remark this answers is not in the document]
Comment (kd4001) [reply src/components/custom/cards/card_deprecated/CardComponentV2.re:10]: Yes

File: src/components/custom/event-tracker/publisher/EventTracker.re [hunk 15]
```diff
@@ -0,0 +1,28 @@
+let getGaEventPayload =
```
[the remark this answers is not in the document]
Comment (kd4001) [reply src/components/custom/event-tracker/publisher/EventTracker.re:1]: If the changes are to me made in future he can make in this repo only and test it in spotcrm with version bump.

File: src/components/generic/accordion/CrmAccordianStatusLabel.re [hunk 16]
```diff
@@ -0,0 +1,14 @@
+let getStatusClassName = (status: CrmAccordionStatus.t): string => {
```
[the remark this answers is not in the document]
Comment (kd4001) [reply src/components/generic/accordion/CrmAccordianStatusLabel.re:1]: This need to be refactored. This is CRM specific code but it is has been put as Accordion component part. 
Weird !!!!!
Will refactor this.

File: src/components/generic/form/LabelV4.re [hunk 17]
```diff
@@ -0,0 +1,4 @@
+[@react.component]
```
[the remark this answers is not in the document]
Comment (kd4001) [reply src/components/generic/form/LabelV4.re:1]: Not required in spotCRM. Please verify that it's not used in fronted-website or virality or any other

File: src/components/styles/abstract/_colorPalette.scss (hunk 1 above)
Comment (kd4001) [reply src/components/styles/abstract/_colorPalette.scss:18]: I think these are porter theme based which Rajiv uses. Any suggestions for changes?

File: src/components/styles/base/_base.scss (hunk 3 above)
Comment (kd4001) [reply src/components/styles/base/_base.scss:20]: Breaking changes. Any suggestions how to remove this ?

File: src/components/styles/components/_header.scss (hunk 5 above)
Comment (kd4001) [reply src/components/styles/components/_header.scss:22]: This is for having common porter banner.

File: src/components/styles/main.scss (hunk 6 above)
Comment (kd4001) [reply src/components/styles/main.scss:1]: All the scss in imported to single main.scss and who ever wants to use any mixin or any property will import it..
Kind of blunder.
Will be moving component specific css to component folder and will be imported there.
_colorPallete, mixins, variable, utilities will still be imported to a single scss file and then to use any of the these you need to have single import.

File: src/entities/OptionList.re (hunk 7 above)
Comment (kd4001) [reply src/entities/OptionList.re:1]: Used for termination and suspension reasons json decoder types. Legacy code.

File: src/utils/ArrayUtils.re (hunk 11 above)
Comment (kd4001) [reply src/utils/ArrayUtils.re:1]: Yes

File: src/components/styles/base/_utilities.scss (hunk 4 above)
Comment (Prateek13727) [reply src/components/styles/base/_utilities.scss:270]: we can address this once we refactor css.

File: src/components/styles/abstract/_colorPalette.scss (hunk 1 above)
Comment (Prateek13727) [reply src/components/styles/abstract/_colorPalette.scss:18]: These are generic color themes Ashutosh. going forward we will use/add the colors to this palette only for consistency

File: src/components/styles/abstract/_mixins.scss (hunk 2 above)
Comment (Prateek13727) [reply src/components/styles/abstract/_mixins.scss:15]: any suggestions for the names ?

File: src/utils/FormValidationUtils.re (hunk 12 above)
Comment (Prateek13727) [reply src/utils/FormValidationUtils.re:28]: @kd4001 ek baar check kar le if we can use a single logic for mobile number validation

File: src/helper-modules/sentry/Sentry.js (hunk 10 above)
Comment (Prateek13727) [reply src/helper-modules/sentry/Sentry.js:9]: @kd4001 take the update from master for this file. udhar updated hay

Comment (kd4001) [reply src/helper-modules/sentry/Sentry.js:9]: okay

File: src/components/custom/error/ErrorDisplayComponentV2.re [hunk 18]
```diff
@@ -0,0 +1,14 @@
+[@react.component]
```
[the remark this answers is not in the document]
Comment (Prateek13727) [reply src/components/custom/error/ErrorDisplayComponentV2.re]: Let's remove ErrorComponent altogether. @ashutheone we may need to test 2 RnR pages

File: src/components/custom/event-tracker/publisher/EventTracker.re (hunk 15 above)
[the remark this answers is not in the document]
Comment (Prateek13727) [reply src/components/custom/event-tracker/publisher/EventTracker.re:1]: We can make new changes on top of it

**Pull request merged**